### PR TITLE
Various

### PR DESCRIPTION
**Pull request by dandehavilland**
Supersedes #42 

---

- Reorganize modules for cleaner lookup, avoid all completions defaulting to plant completions
- Update for API changes `Room` -> `Location`
- Fixes to get connected with Metrc sandbox

### CODE REVIEW

File: app/services/integration_service.rb [hunk 1]
```diff
@@ -30,4 +30,8 @@ def call
       end
     end
   end
+
+  def self.vendor_module(integration)
+    "#{integration.vendor.camelize}Service::Batch".constantize
```
Comment (betacar) [inline app/services/integration_service.rb]: Is it correct that the `Batch` completion handler is hardcoded here?

**Comment by dandehavilland** [comment]
Non-crop completions were triggering crop completions (e.g. sales order start completion would trigger a crop start handler), so the refactor and file tree reorg resolves this.

Also the VendorJob now looks up the correct completion using the same code as the Batch handler instead of being hard coded.

File: app/services/integration_service.rb (hunk 1 above)
Comment (dandehavilland) [reply app/services/integration_service.rb]: I think this is redundant now. I'll confirm later and remove if so.

**Comment by dandehavilland** [comment]
Also incorporates a couple of portal API changes that came in over the weekend (includes on batch for harvests and discard now cause 500)

File: app/services/integration_service.rb (hunk 1 above)
Comment (dandehavilland) [reply app/services/integration_service.rb]: Confirmed & removed.

**Pull request merged**